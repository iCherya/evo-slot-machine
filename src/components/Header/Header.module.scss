@import '@/styles/variables.scss';

.wrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.title {
  font-size: $font-size-large;
  font-weight: $font-weight-bold;
}
