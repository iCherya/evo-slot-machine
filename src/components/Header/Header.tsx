import React from 'react';

export const Header: React.FC = () => {
  return <div>Header component</div>;
};
