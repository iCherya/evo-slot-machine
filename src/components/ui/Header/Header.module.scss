@import '@/styles/variables.scss';

.header {
  margin: 0 auto;
  width: 75%;
  padding: $spacing-base;
  border-radius: 50%;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
  text-align: center;
  font-size: $font-size-xlarge;
  font-weight: $font-weight-base * 2.2;
  letter-spacing: 0.2rem;
  color: $color-accents-dark;
  text-shadow: 1px 2px 1px $color-base-dark;
  background-color: $color-gold;
  border-left: $spacing-base solid $color-gold;
  border-right: $spacing-base solid $color-gold;
  animation: bumping 1s infinite;
  animation-delay: 0.3s;
  border-width: 0;
}

.inGame {
  animation: none;
  background-color: $color-base-light;
  border-left-color: $color-base-light;
  border-right-color: $color-base-light;
  color: $color-gold;
  text-shadow: 1px 2px 1px $color-base-dark;
}

@keyframes bumping {
  0%,
  40%,
  100% {
    transform: translateY(2px);
  }
  20% {
    transform: translateY(0);
  }
}

@media screen and (max-width: $breakpoint-sm) {
  .header {
    font-size: $font-size-large;
  }
}
