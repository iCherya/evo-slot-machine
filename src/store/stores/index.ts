import { AppStore } from './AppStore';

class RootStore {
  public app: AppStore;

  public constructor() {
    this.app = new AppStore();
  }
}

export type RootStoreType = RootStore;
export const rootStore = new RootStore();
