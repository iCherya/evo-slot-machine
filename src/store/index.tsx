import * as React from 'react';

import { RootStoreType, rootStore } from './stores';

import type { IProviderProps } from '@/types';

export const StoreContext = React.createContext<RootStoreType | null>(null);

export const StoreProvider: React.FC<IProviderProps> = ({ children }) => (
  <StoreContext.Provider value={rootStore}>{children}</StoreContext.Provider>
);

export const useStore = (): RootStoreType => {
  const store = React.useContext(StoreContext);

  if (!store) {
    throw new Error('useStore must be used within a StoreProvider.');
  }

  return store;
};
