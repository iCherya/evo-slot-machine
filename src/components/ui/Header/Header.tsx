import React, { useCallback } from 'react';
import classNames from 'classnames';
import { observer } from 'mobx-react-lite';

import { useStore } from '@/store';
import { CONTENT } from '@/config/content';
import { AnimatedNumber } from '@/components/logic/AnimatedNumber';

import styles from './Header.module.scss';

const dummy = {
  user: {
    balance: 42,
  },
};

export const Header: React.FC = observer(() => {
  const { game } = useStore();
  const { user } = dummy;
  const {
    ui: { header },
    domain: { currency },
  } = CONTENT;

  const format = useCallback((number: number) => `${Math.floor(number)} ${currency}`, [currency]);

  return (
    <header data-testid="header" className={classNames(styles.wrapper, game.isStarted && styles.game)}>
      {!game.isStarted ? header.title : <AnimatedNumber number={user.balance} format={format} />}
    </header>
  );
});
