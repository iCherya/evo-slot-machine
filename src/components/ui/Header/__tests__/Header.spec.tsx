import { render } from '@testing-library/react';

import { Header } from '@/components/ui/Header';
import { useStore } from '@/store';

const mockedUseStore = useStore as jest.Mock;

jest.mock('@/store', () => ({
  useStore: jest.fn(),
}));

jest.mock('@/components/logic/AnimatedNumber', () => ({
  AnimatedNumber: jest.fn(({ number }) => <div data-testid="animated-number">{number}</div>),
}));

jest.mock('@/components/logic/Translations', () => ({
  // eslint-disable-next-line @typescript-eslint/member-delimiter-style
  TranslateText: ({ translationKey }: { translationKey: string }) => <span>{translationKey}</span>,
}));

describe('Header', () => {
  it('should render header title if game is not started', () => {
    mockedUseStore.mockImplementationOnce(() => ({
      game: {
        isStarted: false,
      },
    }));

    const { getByTestId } = render(<Header />);
    const header = getByTestId('header');

    expect(header.textContent).toBe('ui.header.title');

    expect(header).toMatchSnapshot();
  });

  it('should render user balance with currency if game is started', () => {
    mockedUseStore.mockImplementationOnce(() => ({
      game: {
        isStarted: true,
      },
      user: {
        balance: 42,
      },
    }));

    const { getByTestId } = render(<Header />);
    const header = getByTestId('header');
    const animatedNumber = getByTestId('animated-number');

    expect(animatedNumber.textContent).toBe('42');

    expect(header).toMatchSnapshot();
  });
});
