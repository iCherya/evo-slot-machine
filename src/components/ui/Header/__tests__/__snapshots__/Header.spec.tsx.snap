// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header title if game is not started 1`] = `
<header
  class=""
  data-testid="header"
>
  JACKPOT
</header>
`;

exports[`Header should render user balance with currency if game is started 1`] = `
<header
  class=""
  data-testid="header"
>
  0 â‚¬
</header>
`;
